pkgs/by-name/ni/nix-scheduler-hook/package.nix

```
@@ -0,0 +1,69 @@
+{
+  fetchFromGitHub,
+  stdenv,
+  lib,
+  nix,
+  meson,
+  cmake,
+  ninja,
+  boost,
+  pkg-config,
+  nlohmann_json,
+  curl,
+}:
+let
+  restclient-cpp = fetchFromGitHub {
+    owner = "mrtazz";
+    repo = "restclient-cpp";
+    rev = "3356f816b161279cfbe318c45cb07c07fb8de6df";
+    hash = "sha256-9//KssNRD7OJFNFdXgzsu7rKP/Nlb4wtmBjfhOt2Vgw=";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "nix-scheduler-hook";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "lisanna-dettwyler";
+    repo = pname;
+    tag = "v" + version;
+    hash = "sha256-qUFqwIdBKHFmHg0J44YFKdc3/AKWk/UTWOMi8qQ01iU=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    cmake
+    ninja
+    boost
+    pkg-config
+    nlohmann_json
+    curl
+  ];
+
+  buildInputs = [
+    nix.libs.nix-util
+    nix.libs.nix-store
+    nix.libs.nix-main
+  ];
```

```suggestion
  nativeBuildInputs = [
    meson
    cmake
    ninja
    pkg-config  
  ];

  buildInputs = [
    boost
    curl
    nix.libs.nix-util
    nix.libs.nix-store
    nix.libs.nix-main
    nlohmann_json
  ];
```
Do we really need cmake and pkg-config? Usually only of them is required and the warning messages around it can be missleading.

---

restclient-cpp in meson.build is imported as a cmake subproject, so cmake is required. pkg-config is used by meson for finding the nix libs.